server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -79,4 +185,292 @@ public void clusterChanged(ClusterChangedEvent event) {
+            remoteClusters.put(name, remote);
+            remote.ensureConnected(listener);
+        } else if (remote.shouldRebuildConnection(connectionSettings)) {
+            // Changes to connection configuration. Must tear down existing connection
+            try {
+                IOUtils.close(remote);
+            } catch (IOException e) {
+                LOGGER.warn("failed to close remote cluster connections for cluster: " + name, e);
+            }
+            remoteClusters.remove(name);
+            remote = new RemoteClusterConnection(settings, connectionSettings, name, transportService);
+            remoteClusters.put(name, remote);
+            remote.ensureConnected(listener);
+        } else {
+            // No changes to connection configuration.
+            listener.onResponse(null);
+        }
+    }
+
+    public boolean isSubscribedIndex(String indexName) {
+        return subscribedIndices.containsKey(indexName);
+    }
+
+    @Nullable
+    public String subscriptionName(String indexName) {
+        return subscribedIndices.get(indexName);
+    }
+
+    @Nullable
+    public Subscription subscription(String indexName) {
+        var subscriptionName = subscribedIndices.get(indexName);
+        if (subscriptionName != null) {
+            return subscriptionsMetadata.subscription().get(subscriptionName);
+        }
+        return null;
+    }
+
+    public void startReplication(String subscriptionName,
+                                 Subscription subscription,
+                                 ActionListener<AcknowledgedResponse> listener) {
+        getPublicationState(
+            subscriptionName,
+            subscription,
+            new ActionListener<>() {
+
+                @Override
+                public void onResponse(PublicationsStateAction.Response stateResponse) {
+                    verifyTablesDoNotExist(
+                        subscriptionName,
+                        stateResponse,
+                        r -> {
+                            for (var index : stateResponse.concreteIndices()) {
+                                subscribedIndices.put(index, subscriptionName);
+                            }
+                            initiateReplication(subscriptionName, subscription, stateResponse, listener);
+                        },
+                        listener::onFailure
+                    );
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    listener.onFailure(e);
+                }
+            }
+        );
+    }
+
+    public void verifyTablesDoNotExist(String subscriptionName,
+                                       Subscription subscription,
+                                       Consumer<Void> onSuccess,
+                                       Consumer<Exception> onFailure) {
+        getPublicationState(
```

Comment:
The naming here seems a bit odd. It's getting the tables from the publication state and then verifying those. Maybe worth breaking apart?